### PR TITLE
Add production deploy pipeline & smoke test task

### PR DESCRIPTION
**Pull request by rjbaker**
Moves deploy task code into separate, reusable task yml.
Adds production deployment task.
Adds prod smoke test to run after deployment.

### CODE REVIEW

File: concourse/pipeline.yml [hunk 1]
```diff
@@ -35,65 +35,61 @@ jobs:
...
-          platform: linux
-          image_resource:
-            type: docker-image
-            source:
-              repository: governmentpaas/curl-ssl
-          run:
-            # TODO we can come up with a more thorough test than this
-            path: curl
-            args: ['--fail', '--silent', '--verbose', '--user', 'gds:((basic-auth-password))', 'https://govuk-coronavirus-business-volunteer-form-stg.cloudapps.digital/']
+        file: govuk-coronavirus-vulnerable-people-form/concourse/tasks/smoke-test.yml
```
Comment (bilbof) [inline concourse/pipeline.yml]: It looks like this filename needs to be changed

Comment (richardTowers) [inline concourse/pipeline.yml]:
```suggestion
        file: govuk-coronavirus-business-volunteer-form/concourse/tasks/smoke-test.yml
```

File: concourse/pipeline.yml [hunk 2]
```diff
@@ -35,65 +35,61 @@ jobs:
...
+          HOSTNAME: govuk-coronavirus-business-volunteer-form-prod
+          SENTRY_DSN: ((sentry-dsn))
+
+  - name: smoke-test-prod
+    plan:
+      - get: govuk-coronavirus-business-volunteer-form
+        trigger: true
+        passed: [deploy-to-prod]
+      - task: smoke-test
+        file: govuk-coronavirus-vulnerable-people-form/concourse/tasks/smoke-test.yml
```
Comment (bilbof) [inline concourse/pipeline.yml]:
```suggestion
        file: govuk-coronavirus-business-volunteer-form/concourse/tasks/smoke-test.yml
```

Comment (richardTowers) [inline concourse/pipeline.yml]:
```suggestion
        file: govuk-coronavirus-business-volunteer-form/concourse/tasks/smoke-test.yml
```

**Review by theseanything — DISMISSED** [review]

**Review by richardTowers — CHANGES_REQUESTED** [review]
Couple of "vulnerabilities"

File: concourse/pipeline.yml [hunk 3]
```diff
@@ -35,65 +35,61 @@ jobs:
...
+      - get: govuk-coronavirus-business-volunteer-form
+        trigger: true
+        passed: [smoke-test-staging]
+      - task: deploy-to-paas
+        file: govuk-coronavirus-business-volunteer-form/concourse/tasks/deploy-to-govuk-paas.yml
+        params:
+          CF_SPACE: production
+          INSTANCES: 30
+          CF_STARTUP_TIMEOUT: 15 # minutes
+          REQUIRE_BASIC_AUTH:
```
Comment (bilbof) [inline concourse/pipeline.yml:77]: It looks like `REQUIRE_BASIC_AUTH` isn't true for prod. Do we want that?

**Review by bilbof — COMMENTED** [review]
Looks good, just a couple comments

File: concourse/pipeline.yml (hunk 3 above)
Comment (rjbaker) [reply concourse/pipeline.yml:77]: It will default to falsy

Comment (richardTowers) [reply concourse/pipeline.yml:77]: When we went live with the other form we had basic auth on for about 5 mins after we mapped the route. Probably not worth it.

Comment (bilbof) [reply concourse/pipeline.yml:77]: It might be nice to have basic auth enabled if we want to test the DB etc is working in prod before we release.

**Review by richardTowers — APPROVED** [review]

**Pull request merged**